### ISSUE TITLE
Preview doesn't work

### ISSUE THREAD

**Issue by agarrharr** [amiantos/lifesaver#4]
I'm on macOS Mojave 10.14.5

![Screen Shot 2019-06-06 at 11 16 47](https://user-images.githubusercontent.com/3266363/59052618-b231a200-884c-11e9-8b05-6b472bef4897.png)

**Comment by amiantos** [comment]
It seems like this might be a bug in macOS, it affects a lot of screensavers.

I'll see if I need to do what's listed at the bottom of this issue to see if it fixes it: https://github.com/JohnCoates/Aerial/issues/464

**Comment by amiantos** [comment]
Hmm, I'm not sure what's going on here. I already have `NSPrincipalClass` defined, and the other thing I found about it (https://github.com/Wandmalfarbe/Simple-Clock-Screensaver/issues/1#issuecomment-437621510) doesn't seem to apply either.

I have seen this happen with other Swift screensavers on my own machine, like I've seen it happen to https://github.com/pedrommcarrasco/Brooklyn as well.

**Comment by amiantos** [comment]
Interesting Apple developer forums say to not make screensavers using Swift, lol... seems like that's news to everyone. https://forums.developer.apple.com/message/268962#268962

**Comment by TheCloudGuyOrg** [comment]
I posted in https://github.com/JohnCoates/Aerial/issues/464 as well. I am getting this issue on my 2018 MacMini but not my 2017 13" MBP. Both are running macOS Mojave 10.14.5

2018 MacMini (Not Working)
Mac mini (2018)
Processor 3.2 GHz Intel Core i7
Memory 32 GB 2667 MHz DDR4
Graphics Intel UHD Graphics 630 1536 MB

2017 13" MBP (Working)
MacBook Pro (13-inch, 2017, Two Thunderbolt Ports)
Processor 3.2 GHz Intel Core i7
Memory 32 GB 2667 MHz DDR4
Graphics Intel Iris Plus Graphics 640 1536 MB

**Comment by glouel** [comment]
> Interesting Apple developer forums say to not make screensavers using Swift, lol... seems like that's news to everyone. https://forums.developer.apple.com/message/268962#268962

Yeah it's barely supported but it does work(ish). You'll notice that the Screensaver template in Xcode is Objective-C only. Apple seems to be willing to fix this by throwing the whole `.saver` model away, to be replaced by a (yet undocumented) `.appex` application extension for screensavers in Catalina. But as of latest Xcode beta there's no way to create those (that I've seen at least). 

This is a pain because a `.saver` runs in a container now in Catalina, which causes a lot of issues for Aerial (I'm the maintainer there). I have many radars open, with little feedback so far (they only fixed multi monitor support). More here if you are interested: https://github.com/JohnCoates/Aerial/issues/801

In any case, the preview bug is because of the Swift libraries when switching from one Swift screensaver to another, we have this in our FAQ : 

> "You cannot use the Aerial screen saver with this version of macOS." error: Select Aerial, close System Preferences with Aerial still selected, re-open System Preferences and Aerial should now work. This is a known bug with Swift screen savers in macOS/OS X reported (a long time ago...) to Apple as [rdar://25569037](http://www.openradar.me/25569037).

I gave a longer reply here too, hopefully this will help you : https://github.com/JohnCoates/Aerial/issues/464#issuecomment-522270123

**Comment by TheCloudGuyOrg** [comment]
@glouel, I posted the longer reply there as well. Since https://github.com/JohnCoates/Aerial/issues/464#issuecomment-522270123 is the main thread I am going to keep all the details in that case.

**Comment by jeffreypriebe** [comment]
2015 15" MBP (Working)
MacBook Pro (Retina, 15-inch, Mid 205)
Processor 2.2 GHz Intel Core i7
Memory 16 GB 1600 MHz DDR3
Graphics Intel Iris Pro 1536 MB
